include/kernel.h

```diff
@@ -389,7 +389,7 @@ struct _thread_base {
 
 	/* this thread's entry in a ready/wait queue */
 	union {
-		sys_dlist_t qnode_dlist;
+		sys_dnode_t qnode_dnode;
```

Comment:
Minor point, but if you're going to correct naming, it should remain "qnode_dlist" or become "qnode_list" or something.  It's the node field that goes into a list, to distinguish it from the equivalent field that goes into an rbtree.  There's little value in trying to encode the type of an object in its field name.

Reply:
You can bring this up when I re-open #12248 and it shows up there, but basically---no, it's not encoding the type in the name, it's encoding the value's role.

The whole problem is the original name results in passing something named `qnode_dlist` in a position where the API requires a `sys_dnode_t` instance.  It looks weird and is confusing.  The field should not claim to be a list when it's a node.  The list is somewhere else.

Reply:
> no, it's not encoding the type in the name, it's encoding the value's role.

Peter, please don't start this fight.   Variable names don't uniformly encode what a value "is" and never have, that's what compiler typesafety is for (which fell down here because of a design glitch with dlist -- those are typedefs and not separate structs, but that's a digression).  Names tell you what it does.

The type was wrong, that's a bug that you fixed.  The name was fine, and correctly indicates the intent of the author whether or not you personally agree with the convention in use.  It's not productive (or even feasible) to run around giant codebases "fixing" stuff like other people's variable naming, no one ever agrees with that kind of thing and it just makes a mess.  Some fights just aren't worth having.